ui/apps/dashboard/src/app/(auth)/sign-up/account-setup/page.tsx

```diff
@@ -3,38 +3,29 @@
 import { useEffect, useState } from 'react';
 import type { Route } from 'next';
 import { useRouter } from 'next/navigation';
-import { useUser } from '@clerk/nextjs';
+import { useAuth, useOrganization, useOrganizationList, useUser } from '@clerk/nextjs';
 import { ExclamationCircleIcon } from '@heroicons/react/20/solid';
 
+import { useBooleanFlag } from '@/components/FeatureFlags/hooks';
 import LoadingIcon from '@/icons/LoadingIcon';
 
 export default function AccountSetupPage() {
   const router = useRouter();
   const secondsElapsed = useSecondsElapsed();
+  const { getToken } = useAuth();
   const { user } = useUser();
-
-  const isAccountSetup = user && user.externalId && user.publicMetadata.accountID;
-
-  // Reload the user until the account is set up
-  useEffect(() => {
-    if (!user || isAccountSetup) return;
-
-    const intervalID = setInterval(() => {
-      user.reload();
-    }, 500);
-
-    return () => {
-      clearInterval(intervalID);
-    };
-  }, [isAccountSetup, user]);
+  useForceActiveOrganization();
+  const isAccountSetup = useIsAccountSetup();
 
   // Redirect to the home page once the account is set up
   useEffect(() => {
     if (!isAccountSetup) return;
 
-    // We use `replace` so that the user doesn't get redirected back to this page if they click the back button
```

Comment:
Nit: This is a super useful comment, can we keep it?

Reply:
Removed by accident. Re-adding it